sci-libs/opencascade/opencascade-7.5.2.ebuild

```diff
@@ -0,0 +1,175 @@
+	freeimage? ( media-libs/freeimage )
+	json? ( dev-libs/rapidjson )
+	tbb? ( dev-cpp/tbb )
+	vtk? ( sci-libs/vtk[rendering] )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	doc? ( app-doc/doxygen )
+	examples? ( dev-qt/linguist-tools:5 )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-7.5.1-0001-allow-default-search-path-for-Qt5.patch
+	"${FILESDIR}"/${PN}-7.5.1-0002-remove-unnecessary-Qt5-check.patch
+	"${FILESDIR}"/${PN}-7.5.1-0003-add-Gentoo-configuration-type.patch
+	"${FILESDIR}"/${PN}-7.5.1-0004-fix-installation-of-cmake-config-files.patch
+	"${FILESDIR}"/${PN}-7.5.1-0005-fix-write-permissions-on-scripts.patch
+	"${FILESDIR}"/${PN}-7.5.1-0006-fix-creation-of-custom.sh-script.patch
+	"${FILESDIR}"/${PN}-7.5.1-fix-AllValues-name-collision-with-vtk-9.0.patch
+)
+
+src_prepare() {
+	cmake_src_prepare
+
+	if use debug; then
+		append-cppflags -DDEBUG
+		append-flags -g
+	fi
+
+	sed -e 's/\/lib\$/\/'$(get_libdir)'\$/' \
+		-i adm/templates/OpenCASCADEConfig.cmake.in || die
+
+	# There is an OCCT_UPDATE_TARGET_FILE cmake macro that fails due to some
+	# assumptions it makes about installation paths. Rather than fixing it, just
+	# get rid of the mechanism altogether - its purpose is to allow a
+	# side-by-side installation of release and debug libraries.
+	sed -e 's|\\${OCCT_INSTALL_BIN_LETTER}||' \
+		-i "adm/cmake/occt_toolkit.cmake" || die
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_DOC_Overview=$(usex doc)
+		-DBUILD_Inspector=$(usex examples)
+		-DBUILD_WITH_DEBUG=$(usex debug)
+		-DCMAKE_INSTALL_PREFIX="/usr"
+		-DINSTALL_DIR_BIN="$(get_libdir)/${P}/bin"
+		-DINSTALL_DIR_CMAKE="$(get_libdir)/cmake/${P}"
+		-DINSTALL_DIR_DOC="share/doc/${PF}"
+		-DINSTALL_DIR_LIB="$(get_libdir)/${P}"
+		-DINSTALL_DIR_SCRIPT="$(get_libdir)/${P}/bin"
+		-DINSTALL_DIR_WITH_VERSION=ON
+		-DINSTALL_SAMPLES=$(usex examples)
+		-DINSTALL_TEST_CASES=NO
+		-DUSE_D3D=NO
+		-DUSE_FFMPEG=$(usex ffmpeg)
+		-DUSE_FREEIMAGE=$(usex freeimage)
+		-DUSE_FREETYPE=ON
+		-DUSE_GLES2=$(usex gles2)
+		-DUSE_RAPIDJSON=$(usex json)
+		-DUSE_TBB=$(usex tbb)
+		-DUSE_VTK=$(usex vtk)
+	)
+
+	use doc && mycmakeargs+=( -DINSTALL_DOC_Overview=ON )
+
+	if use examples; then
+		mycmakeargs+=(
+			-D3RDPARTY_QT_DIR="${ESYSROOT}"/usr
+			-DBUILD_SAMPLES_QT=ON
+		)
+	fi
+
+	if use vtk; then
+		if has_version ">=sci-libs/vtk-9.0.0"; then
+			mycmakeargs+=(
+				-D3RDPARTY_VTK_DIR="${ESYSROOT}"/usr
+				-D3RDPARTY_VTK_INCLUDE_DIR="${ESYSROOT}"/usr/include/vtk-9.0
+				-D3RDPARTY_VTK_LIBRARY_DIR="${ESYSROOT}"/usr/$(get_libdir)
+			)
+		fi
+	fi
+
+	cmake_src_configure
+
+	prepare_env_file() {
+		# prepare /etc/env.d file
+		sed -e 's|VAR_CASROOT|'${ESYSROOT}'/usr|g' < "${FILESDIR}/${PN}-${PV_MAJ}.env.in" >> "${T}/${PV_MAJ}" || die
+		sed -e 's|lib/|'$(get_libdir)'/|g' -i "${T}/${PV_MAJ}" || die
+		sed -e 's|VAR_PV|'${PV}'|g' -i "${T}/${PV_MAJ}" || die
+
+		# use TBB for memory allocation optimizations?
+		use tbb && (sed -e 's|^#MMGT_OPT=0$|MMGT_OPT=2|' -i "${T}/${PV_MAJ}" || die)
```

Comment:
> Since I joined proxy-maint earlier, might as well start with this.
> 
> While essentially same as previous ebuild, there's a few things that bother me, be nice if could improve some of it (along with optix, sorry for being picky again :)

There's nothing to say about being picky :) But please stick to the point and don't mix with personal preferences.

And congrats on joining proxy-maint! I think, it's a good addition to the team.